pkg/config/analysis/analyzers/testdata/gateway-custom-ingressgateway-badport.yaml

```diff
@@ -9,7 +9,7 @@ spec:
     myapp: private-ingressgateway
   servers:
   - port:
-      number: 80
+      number: 8004
```

Comment:
Can you also add a case, when there's no service targetPort, and the service port is not matching the gateway port? This will also be a no matching case since the listener will be the gateway port.

Reply:
Sure, I will add one

Reply:
Done

Reply:
Thanks! Looking good